src/s2geography/geoarrow_test.cc

```diff
@@ -202,3 +208,256 @@ TEST(GeoArrow, GeoArrowReaderReadWKTCollection) {
+  EXPECT_GT(ys[4], 45);
+
+  // with tessellation and projection
+  nanoarrow::UniqueArray array3;
+  options.set_projection(s2geography::pseudo_mercator());
+  writer.Init(schema.get(), options);
+  writer.WriteGeography(*geog);
+  writer.Finish(array3.get());
+
+  EXPECT_EQ(array3->length, 1);
+  EXPECT_GT(array3->children[0]->length, 2);
+  EXPECT_EQ(array3->children[0]->length, 9);
+
+  xs = reinterpret_cast<const double*>(array3->children[0]->children[0]->buffers[1]);
+  EXPECT_NEAR(xs[0], -7124447.41, 0.01);
+  EXPECT_DOUBLE_EQ(xs[9], 0);
+  ys = reinterpret_cast<const double*>(array3->children[0]->children[1]->buffers[1]);
+  EXPECT_NEAR(ys[0], 5621521.48, 0.01);
+  EXPECT_NEAR(ys[8], 5621521.48, 0.01);
+  EXPECT_GT(ys[4], 5621521);
+}
+
+TEST(GeoArrow, GeoArrowWriterPolygonTessellated) {
+  s2geography::WKTReader reader;
+  auto geog = reader.read_feature("POLYGON ((-64 45, 0 45, 0 55, -64 55, -64 45))");
+
+  Writer writer;
+  nanoarrow::UniqueSchema schema;
+  GeoArrowSchemaInitExtension(schema.get(), GEOARROW_TYPE_WKT);
+
+  nanoarrow::UniqueArray array;
+  writer.Init(schema.get());
+  writer.WriteGeography(*geog);
+  writer.Finish(array.get());
+
+  EXPECT_EQ(array->length, 1);
+  auto length_no_tesselation = reinterpret_cast<const int32*>(array->buffers[1])[1];
+
+  // with tessellation -> more coordinates
+  nanoarrow::UniqueArray array2;
+  s2geography::geoarrow::ExportOptions options;
+  options.set_tessellate_tolerance(S1Angle::Radians(10000 / EARTH_RADIUS_METERS));
+
+  writer.Init(schema.get(), options);
+  writer.WriteGeography(*geog);
+  writer.Finish(array2.get());
+
+  EXPECT_EQ(array2->length, 1);
+  auto length_with_tesselation = reinterpret_cast<const int32*>(array2->buffers[1])[1];
+
+  // dummy test to check that the WKT string length is larger with tesselation
+  EXPECT_GT(length_with_tesselation, length_no_tesselation);
+}
+
+void TestGeoArrowRoundTrip(s2geography::Geography& geog, GeoArrowType type) {
+  // writing
+  Writer writer;
+  nanoarrow::UniqueSchema schema;
+  nanoarrow::UniqueArray array;
+
+  GeoArrowSchemaInitExtension(schema.get(), type);
+  writer.Init(schema.get());
+  writer.WriteGeography(geog);
+  writer.Finish(array.get());
+
+  EXPECT_EQ(array->length, 1);
+
+  // reading back
+  Reader reader;
+  std::vector<std::unique_ptr<s2geography::Geography>> result;
+
+  reader.Init(schema.get());
+  reader.ReadGeography(array.get(), 0, array->length, &result);
+
+  EXPECT_EQ(result[0]->dimension(), geog.dimension());
+  EXPECT_EQ(result[0]->num_shapes(), geog.num_shapes());
+  // TODO better assert equal for the geographies
+}
+
+TEST(GeoArrow, GeoArrowRoundtripPoint) {
+  s2geography::WKTReader reader;
+  auto geog = reader.read_feature("POINT (30 10)");
+
+  TestGeoArrowRoundTrip(*geog, GEOARROW_TYPE_POINT);
+  TestGeoArrowRoundTrip(*geog, GEOARROW_TYPE_INTERLEAVED_POINT);
+  TestGeoArrowRoundTrip(*geog, GEOARROW_TYPE_WKT);
+  TestGeoArrowRoundTrip(*geog, GEOARROW_TYPE_WKB);
+}
```

Comment:
Added basic roundtrip tests like this for all geometry types. We can certainly improve that further (e.g. by improving the helper to directly take the strings, or work with a vector of geographies so we can directly test multiple point objects (like adding a point empty, or missing value)